lighthouse-core/gather/gatherers/start-url.js

```diff
@@ -49,24 +49,29 @@ class StartUrl extends Gatherer {
   pass(options) {
     return options.driver.getAppManifest()
       .then(response => {
-        return manifestParser(response.data, response.url, options.url);
+        return response && manifestParser(response.data, response.url, options.url);
       })
       .then(manifest => {
-        if (!manifest.value.start_url || !manifest.value.start_url.raw) {
-          return Promise.reject(new Error(`No web app manifest found on page ${options.url}`));
+        if (!manifest || !manifest.value.start_url || !manifest.value.start_url.raw) {
+          throw new Error(`No web app manifest found on page ${options.url}`);
         }
 
         if (manifest.value.start_url.debugString) {
-          return Promise.reject(new Error(manifest.value.start_url.debugString));
+          throw new Error(manifest.value.start_url.debugString);
```

Comment:
do we want this to be an error? It'll be things like `invalid start_url relative to ${manifestUrl}` or `start_url must be same-origin as document`, which is nice to put in front of the developer, but the parsed manifest will still have a valid `start_url` in those cases.

Not sure if we want to fall on the "this is broken" side or the side of how the browser actually treats it since we don't have a great logging spot for that type of message if no error here